File: laba3/main.c
#include <stdio.h>
#include <time.h>

int main() {
    clock_t start_time = clock(); 

    printf("hello world!\n");
    printf("%s\t %s\t %s\n", "Цельсии", "Фаренгейты", "Кельвины");

    for (int i = 0; i <= 100; i += 10) {
        float celsius = (float)i;
        float fahrenheit = (celsius * 9.0 / 5.0) + 32.0;
        float kelvin = celsius + 273.15;

        // Print values with tabulation
        printf("%.1f\t %.1f\t %.1f\n", celsius, fahrenheit, kelvin);
    }

    clock_t end_time = clock();
    double dur_time = (double)(end_time - start_time); 
    printf("Время выполнения: %.6f секунд\n", dur_time);

    return 0;
}

File: laba3/main.rs
use std::time::Instant;

fn main() {
    let start_time = Instant::now(); 

    println!("hello world!");
    println!("{:<10} {:<15} {:<10}", "Цельсии", "Фаренгейты", "Кельвины"); 

    for i in (0..=100).step_by(10) {
        let celsius = i as f32;
        let fahrenheit = (celsius * 9.0 / 5.0) + 32.0;
        let kelvin = celsius + 273.15;

       
        println!("{} \t\t{} \t\t{}", celsius, fahrenheit, kelvin);
    }

    let duration = start_time.elapsed(); 
    let microseconds = duration.as_micros(); 
    println!("Время выполнения: {} микросекунд", microseconds);
}